### PR TITLE
Launchpad: update disabling logic

### PR DESCRIPTION
**Pull request by mattwiebe**
The previous launchpad completion code baked in an unfortunate assumption: that _all tasks_ must be completed in order to deactivate Launchpad. This is incorrect. Our previous logic, [on the client](https://github.com/Automattic/wp-calypso/blob/bda8d93ffcbbb3a2e20e69cc25c858b6ab209ee7/client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts#L469), was to assume that having completed the final task in a list was equivalent to completing Launchpad. 

So this replicates that, and also allows for explicitly marking a task as `isLaunchTask` to say "complete this and Launchpad is done." I recommend that we map this onto existing flows.

## Proposed changes:

* Allows launchpad disabling whenever the last task, or the `isLaunchTask`, is completed.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

With this patch applied to your dotcom sandbox, trying completing a flow without first completing all other steps. With this you should be able to. Without it, you shouldn't be able to.

### CODE REVIEW

File: projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php [hunk 1]
```diff
@@ -390,7 +390,7 @@ public function add_hooks_for_active_tasks( $task_list_id = null ) {
 				// We only need to know the built completion status if the task has an `add_listener_callback` property.
 				// Small optimization to not run `is_complete_callback` as often.
 				$task = $this->build_task( $task_definition );
-				if ( ! $task['completed'] && is_callable( $task_definition['add_listener_callback'] ) ) {
```
Comment (lezama) [inline projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php:397]: we don't want to check `is_callable`?

File: projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php [hunk 2]
```diff
@@ -414,7 +414,17 @@ public function mark_task_complete( $task_id ) {
 		$statuses[ $key ] = true;
 		$result           = update_option( 'launchpad_checklist_tasks_statuses', $statuses );
 
-		$this->maybe_disable_launchpad();
+		if ( isset( $task['isLaunchTask'] ) && $task['isLaunchTask'] ) {
```
Comment (lezama) [inline projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php:421]:
```suggestion
		if ( ! empty( $task['isLaunchTask'] ) ) {
```

File: projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php (hunk 1 above)
Comment (mattwiebe) [reply projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php:397]: It turned out that it was already being checked above it, before we even built the task. It was too late to stop the merge train so I decided to clean it up here. [our discussion](https://github.com/Automattic/jetpack/pull/30527#discussion_r1190399533)

File: projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php (hunk 2 above)
Comment (mattwiebe) [reply projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php:421]: I preferred mine, to guard against the kind of guy (I'm that kind of guy) who would add `isLaunchTask=false` in a task.

Comment (mattwiebe) [reply projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php:421]: Oh right looking at this again I forgot empty also checked for false (yay PHP), I do like that better.

**Comment by mattwiebe** [comment]
I finally circled back to test this and it works as described: completing the last task of the `isLaunchTask` will switch `launchpad_screen` to `off` as previously. Any sites that launch currently are relying on the [client-side check](https://github.com/Automattic/wp-calypso/blob/bda8d93ffcbbb3a2e20e69cc25c858b6ab209ee7/client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts#L469) which isn't great.

 @lezama or @edanzer can you help get this merged?

**Comment by samiff** [comment]
> trying completing a flow without first completing all other steps. With this you should be able to. Without it, you shouldn't be able to.

As someone not all that familiar with Launchpad, could you explain a bit more on the testing aspect of this PR please.

It's not clear to me for example, if I've completed all launch tasks, should the Launchpad and "Launch your site" button still display, or should launch happen automatically when all tasks are completed?

**Pull request closed without merging**